### PR TITLE
JS: Add steps through terminal styling 

### PR DESCRIPTION
**Pull request by erik-krogh**
Inspired by missing support for the highly dependent upon library [`colors`](https://www.npmjs.com/package/colors). 
And I found a few others to model by searching NPM tags. 

They account for a combined [320 million weekly downloads on NPM](https://i.imgur.com/eXJz9GP.jpg). 

[Evaluation looks fine](https://github.com/dsp-testing/erik-krogh-dca/tree/run/colors-nightly-security-extended/reports).   
No change in results.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/Logging.qll [hunk 1]
```diff
@@ -201,3 +201,134 @@ private class DebugLoggerCall extends LoggerCall, API::CallNode {
...
+}
+
+/**
+ * A step through the [`colors`](https://npmjs.org/package/colors) library.
+ * This step ignores the `String.prototype` modifying part of the `colors` library.
+ */
+class ColorsStep extends TaintTracking::SharedTaintStep {
+  override predicate stringManipulationStep(DataFlow::Node pred, DataFlow::Node succ) {
+    exists(API::CallNode call |
+      call = API::moduleImport(["colors", "colors/safe"]).getAMember*().getACall()
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/frameworks/Logging.qll:224]: `colors/safe` as a `SharedTaintStep` 😂 
```suggestion

      call = API::moduleImport(["colors", "colors/safe" /* this variant avoids modifying the prototype methods */ ]).getAMember*().getACall()
```

**Review by esbena — COMMENTED** [review]
LGTM

What do you think about introducing a super class like `StringDecorationStep extends SharedTaintStep`, and let all of these modules extend that? 

The inspiration for this is that I think sanitization must happen before decoration. It is not clear how that could be turned into a query however. Perhaps a taint kind that is unsanitizable would be useful.

**Comment by erik-krogh** [comment]
> What do you think about introducing a super class like `StringDecorationStep extends SharedTaintStep`, and let all of these modules extend that?

I think we achieve almost the same thing by overriding the `stringManipulationStep` predicate in `SharedTaintStep`.  
So I don't think a `StringDecorationStep` class is needed.

**Review by esbena — DISMISSED** [review]

**Review by esbena — APPROVED** [review]

**Review by duongxuantuan154 — COMMENTED** [review]
- [ ]

**Pull request merged**